### PR TITLE
docs: updates BED_MESH_CALIBRATE description

### PR DESCRIPTION
**Pull request by pedrolamas**
Adds the ADAPTIVE and ADAPTIVE_MARGIN parameters to the documentation.

### CODE REVIEW

**Comment by pedrolamas** [comment]
@Arksine @voidtrance I would really appreciate it if you could review this.

File: docs/G-Codes.md [hunk 1]
```diff
@@ -146,15 +146,21 @@ The following commands are available when the
 (also see the [bed mesh guide](Bed_Mesh.md)).
 
 #### BED_MESH_CALIBRATE
-`BED_MESH_CALIBRATE [METHOD=manual] [HORIZONTAL_MOVE_Z=<value>]
-[<probe_parameter>=<value>] [<mesh_parameter>=<value>]`: This command probes
-the bed using generated points specified by the parameters in the config. After
-probing, a mesh is generated and z-movement is adjusted according to the mesh.
+`BED_MESH_CALIBRATE [PROFILE=<name>] [METHOD=manual] [HORIZONTAL_MOVE_Z=<value>]
+[<probe_parameter>=<value>] [<mesh_parameter>=<value>] [ADAPTIVE=1]
```
[the remark this answers is not in the document]
Comment (pedrolamas) [reply docs/G-Codes.md:150]: Indeed, and the only reason I only specified `ADAPTIVE=1` was to follow the same pattern as with `METHOD=manual` (where the default `METHOD=automatic` is also omitted, and thus, I omitted the default `ADAPTIVE=0`)

File: docs/G-Codes.md [hunk 2]
```diff
@@ -146,15 +146,21 @@ The following commands are available when the
...
+and z-movement is adjusted according to the mesh.
+The mesh will be saved into a profile specified by the `PROFILE` parameter,
+or `default` if unspecified.
 See the PROBE command for details on the optional probe parameters. If
 METHOD=manual is specified then the manual probing tool is activated - see the
 MANUAL_PROBE command above for details on the additional commands available
 while this tool is active. The optional `HORIZONTAL_MOVE_Z` value overrides the
-`horizontal_move_z` option specified in the config file.
+`horizontal_move_z` option specified in the config file. If ADAPTIVE=1 is
+specified then the probed area will use the objects defined by the Gcode file
```
[the remark this answers is not in the document]
Comment (pedrolamas) [reply docs/G-Codes.md:161]: How about this:

> If ADAPTIVE=1 is specified then the objects defined by the Gcode file being printed will be used to define the probed area.

**Review by voidtrance — APPROVED** [review]

**Comment by KevinOConnor** [comment]
Thanks.

-Kevin

**Pull request merged**